Add new package: nftables-slim/1.1.3

yxun opened the pull request:
Ad new package: nftables-slim/1.1.3

This PR is related to a discussion in https://kubernetes.slack.com/archives/C03MASRQP4M/p1747869530558059
I updated the existing nftables configure options in a new yaml file and removed some unused built-time dependencies.
The goal of this new package is to provide a slim package so that users can use nftables via a control process and no interactive user usage.

Thanks,

### Pre-review Checklist

<!--
This checklist is mostly useful as a reminder of small things that can easily be
forgotten – it is meant as a helpful tool rather than hoops to jump through.

At the moment of this PR you have the most information on what all the change
will affect, so please take the time to jot it down.

Put an `x` in all the items that apply, make notes next to any that haven't been
addressed, and remove any items that are not relevant to this PR.

-->

#### For new package PRs only
<!-- remove if unrelated -->
- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [ ] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [ ] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For version bump PRs
<!-- remove if unrelated -->
- [ ] The `epoch` field is reset to 0

yxun committed:
Ad new package: nftables-slim/1.1.3

Signed-off-by: Yuanlin Xu <yuanlin.xu@redhat.com>

yxun commented:
cc @sridhargaddam

skitt commented on nftables-slim.yaml:
```diff
@@ -0,0 +1,58 @@
+  - uses: autoconf/configure
+    with:
+      opts: |
+        --sbindir=/usr/bin --without-cli --disable-man-doc --disable-debug --without-PACKAGE
```
The last flag is extraneous:

```suggestion
        --sbindir=/usr/bin --without-cli --disable-man-doc --disable-debug
```

yxun committed:
review update

Signed-off-by: Yuanlin Xu <yuanlin.xu@redhat.com>

yxun commented:
The [elastic-build](https://github.com/wolfi-dev/os/pull/58274/checks?check_run_id=45322336093) job is failing in Tests. When I checked the log , it shows permission denied from my side.

It looks like an environment issue. I got all job passed before merging latest main branch. And then this job was failing so I force-pushed a revert of the main branch merge.

yxun commented:
cc @imjasonh

yxun commented:
The test commit was for testing purpose. It copies and pastes the exactly same content from the `nftables.yaml` to the new `nftable-slim.yaml`. That was for verifying the CI job elastic-build and for checking if that CI job can pass the original `nftables.yaml` build and test.

yxun committed:
fix lint

Signed-off-by: Yuanlin Xu <yuanlin.xu@redhat.com>

yxun committed:
disable test

Signed-off-by: Yuanlin Xu <yuanlin.xu@redhat.com>

yxun committed:
fix lint error

Signed-off-by: Yuanlin Xu <yuanlin.xu@redhat.com>

sridhargaddam commented:
The Lint jobs seems to have got stuck. Is there a way to re-trigger the tests?

sridhargaddam commented:
@dnegreira we're currently blocked on this PR for adding Nftables support in the Istio project. Could you please take a look when you get a chance, thank you.

yxun committed:
Merge branch 'main' into nftables-slim

yxun committed:
Use builtin mini-gmp

Signed-off-by: Yuanlin Xu <yuanlin.xu@redhat.com>

yxun committed:
remove unused dependencies

Signed-off-by: Yuanlin Xu <yuanlin.xu@redhat.com>

yxun committed:
remove example doc files

Signed-off-by: Yuanlin Xu <yuanlin.xu@redhat.com>

yxun committed:
remove example doc files in make target

Signed-off-by: Yuanlin Xu <yuanlin.xu@redhat.com>

yxun commented:
This PR is ready for review and could I get a trigger of the two github action lint jobs ?
It looks like after I pushed new commits , those two github action lint jobs need a manual trigger. Thanks.

yxun committed:
Fix Lint job error

Signed-off-by: Yuanlin Xu <yuanlin.xu@redhat.com>

kranurag7 reviewed (commented):
thank you for the work here; can you please add some tests here so that the functionality is intact in b/w package version updates.

yxun commented:
> thank you for the work here; can you please add some tests here so that the functionality is intact in b/w package version updates.

@kranurag7 Thank you for checking . Has the CI job been fixed ?
ref: https://github.com/wolfi-dev/os/pull/58487
In the past, the CI job was flaky in the existing nftables.yaml test pipeline . See the ref PR above.

I can add that test step in my PR if that CI job has been fixed.

yxun committed:
Add test pipeline step back

Signed-off-by: Yuanlin Xu <yuanlin.xu@redhat.com>

yxun commented:
I added the test pipeline back in my last commit. 
In the past, Jason and I were blocked by the CI job elastic-build. It was showing false negative test result even rebuilding the existing nftables.yaml package in wolfi-os/nftables.yaml.  Jason told me that he got no error in his local environment and he created a DNM PR for testing that: https://github.com/wolfi-dev/os/pull/58487

@kranurag7 that's why I removed the test pipeline part in my PR.

yxun committed:
Revert "Add test pipeline step back"

This reverts commit d4b94c8ac2d7227068be5abb62c101bb8a59203f.

yxun committed:
Reapply "Add test pipeline step back"

This reverts commit 623a17c5a0f12b6829750cd7430f7a94cdcd5f1f.

yxun committed:
Merge branch 'main' into nftables-slim

yxun commented:
Thanks @kranurag7 . I will try your suggested make steps above around my day time tomorrow.

> I'm not seeing any decrement in size in the slim package so if you motivation was to get a slimmer version of this package then I think we're not reaching there.

The initial concern was the dependencies when installing the existing nftables.yaml package. It pulls many other dependencies and additional unused binary files. 
That causes an Istio side wolfi based image from 10+MB to 30+MB when installing the existing nftables.yaml package.
We only need that nft package which is expected around 10+MB.  

I included some configure changes in this PR. So additional unused binary files and example doc files will not be included.
That will fix an unexpected file checking test from an Istio side job as well.
I can update the new file name to avoid any confusion tomorrow. Thanks.

kranurag7 commented:
nothing on your side @yxun we're trying to land some changes to qemu runner so that tests will pass. I understand and ack your point of nftables pulling in some extra deps at runtime than `nftables-slim`

thank you; I'll keep you updated.

stevebeattie committed:
Merge branch 'main' into nftables-slim

kranurag7 committed:
Merge branch 'main' into nftables-slim

yxun commented:
Hello @imjasonh , @kranurag7 ,
Is there any question from reviewer side about this PR ? Could I ask for a review and approval ? Thanks.

kranurag7 committed:
Merge branch 'main' into nftables-slim

yxun committed:
Remove unused comment

Signed-off-by: Yuanlin Xu <yuanlin.xu@redhat.com>

yxun committed:
Merge branch 'main' into nftables-slim

yxun commented:
Are we waiting for some main branch updates ? I see we have several branch main merge commits in this PR commit history now.

The github actions Lint and Wolfictl Lint still need a manual trigger when we added branch main merge in this PR. 
Should we avoid new merge(s) in this PR so that we don't need re-trigger and rerun those two passed github actions ?

Thanks,

stevebeattie committed:
nftables-slim: fix yam formatting whinge

Signed-off-by: Steve Beattie <steve.beattie@chainguard.dev>

stevebeattie committed:
Merge branch 'main' into nftables-slim

yxun commented:
Hello @kranurag7 , thanks for your help fixing the CI jobs. I see all jobs are passing in this PR now. Could I have a review approval ? Is there anything pending for this PR ?

Thanks,

kranurag7 reviewed (approved):
thanks @yxun this looks good to me.

stevebeattie commented:
This also looks good to me, merging. Thanks!
